### DISCUSSION TITLE
Creating feature classes with both Multi and Single geometry types

### DISCUSSION THREAD

**Discussion by CF-FHB-X** [realiii/fudgeo#11] [Q&A] [answered]
Afternoon!

Quick question regarding creating feature classes with a mixture of both single and multi geometries (i.e. loading a country dataset where some countries are single polygons, whereas others are multi-polygons). What should a user set the `shape_type` attribute to in that case?

```python
fc = gpkg.create_feature_class(
    'test', srs, fields=fields, shape_type=GeometryType.polygon
)
```
or
```python
fc = gpkg.create_feature_class(
    'test', srs, fields=fields, shape_type=GeometryType.multi_polygon
)
```

Additionally, assuming this is something we can do, I assume there's no issue passing a list of mixed-type geometries to the `cursor.executemany` INSERT statement afterwards?

Thanks!
Richard

**Comment by jlhumber** [comment, marked as answer]
You could use either approach above but would not recommend trying to mix geometry types, probably depends on your use case which you choose.  
- If you used `GeometryType.polygon` it would be one polygon for each part of a country (e.g. the USA would be a record for Lower 48, a record for Hawaii (each island), a record for Alaska, ... and so on).  
- If you used `GeometryType.multi_polygon` then each record you insert needs to be a `MultiPolygon`, even is there is just one polygon you'd still make it a `MultiPolygon` (look at `test_multi_polygon` for an example of creating a `MultiPolygon`)

**Reply by CF-FHB-X** [reply]
I think Option 2 is probably best, otherwise we'd end up with ~300 individual rows for a country like Canada, thanks to all the Arctic islands up north.

I was able to confirm that ArcGIS Pro seems to use the same methodology (all single and multi-polygons are written out as multi-polygons). 

There was some discussion on the [GeoPackage GitHub page](https://github.com/opengeospatial/geopackage/issues/525) regarding marking those as type `GEOMETRY` (based on the spec's [requirement](http://www.geopackage.org/spec121/#r32) stating that geometries must be of or a subtype of what's registered in the `gpkg_geometry_columns` table). Not sure if `POLYGON` is a subtype of `MULTIPOLYGON`, or vice versa, but seems `GEOMETRY` is a bit over-arching.

At any rate, converting everything to MULTIPOLYGON seems easy enough (and supported in Pro)!

Thanks!
